src/nxt_router.c

```diff
@@ -1981,6 +1979,13 @@ nxt_router_conf_create(nxt_task_t *task, nxt_router_temp_conf_t *tmcf,
 
             nxt_debug(task, "application: %V", &lscf.application);
 
+            tmcf->listener_conf = &lscf;
+
+            skcf = nxt_router_socket_conf(task, tmcf, &name);
```

Comment:
It looks like your previous way is better, could you try it again?
```
nxt_router_socket_conf(task, tmcf, &name, lscf.backlog);
```
That we don't need to add `listener_conf` to `tmcf`.
I guess it's more concise.

Reply:
Ah, a mixture of the original _and_ this way, yeah, that maybe makes sense.